packages/stream-caip10-link/src/caip10-link.ts

```diff
@@ -122,7 +122,11 @@ export class Caip10Link extends Stream {
      * @param accountId
      */
     static makeGenesis(accountId: AccountID): GenesisCommit {
-        return { header: { controllers: [accountId.toString()],
+        // Ethereum addresses specifically are sometimes encoded with mixed case and
+        // sometimes all lower case. In order to deal with this and not have different
+        // links for different addresses.
+        const controller = accountId.toString().toLowerCase()
```

Comment:
how confident are we that no blockchains will ever have case-sensitive addresses?

It would be a pretty hairy decision to make as a blockchain designer, so here's hoping it never happens...

Reply:
I'm pretty sure for bitcoin this is not the case: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md

I do think we need a better strategy mid/long term. Alternatively we add specific logic for each blockchain here, which is not ideal

Reply:
Yeah wait, the CAIP10 spec specifically says that the accountId is case-senstive.  Blindly translating it to lower case feels wrong then and like it's corrupting the links.  If eth addresses are case insensitive, then that needs to be handled in an eth-specific way, not blindly applied to all possible Caip10Links.

Reply:
Ok, we should do it properly and encode eth addresses as specified by [EIP-55](https://eips.ethereum.org/EIPS/eip-55).

Reply:
On second thought, doing EIP-55 could potentially cause issues for 3IDs which have already been migrated/created on 3ID Connect. Would like @zachferland to weigh in here if this matters or not.

Reply:
> On second thought, doing EIP-55 could potentially cause issues for 3IDs which have already been migrated/created on 3ID Connect. Would like @zachferland to weigh in here if this matters or not.

holding off on final review/approval of this PR until this question is addressed.  It does seem like using EIP-55 encoding would be the "better" solution than just lower-casing it (as https://github.com/pedrouid/caip-js/issues/11 is probably the best long-term solution and I imagine that that's what it would do). But if that breaks 3id-connect that's obviously a problem

Reply:
After a conversation with @zachferland we decided that we should go for the EIP-55 path. Will be updating this PR shortly.